internal/integrations/rawtemplate.go

```diff
@@ -0,0 +1,243 @@
+	for _, pipeline := range pipelines {
+		list = append(list, listIDs(pipeline)...)
+	}
+	slices.SortFunc(list, func(a, b component.ID) int { return strings.Compare(a.String(), b.String()) })
+	return slices.Compact(list)
+}
+
+// listReceivers lists the IDs of all the receivers found in pipelines
+func listReceivers(pipelines map[pipeline.ID]PipelineConfig) []component.ID {
+	return listComponents(pipelines, func(c PipelineConfig) []component.ID {
+		return []component.ID{*c.Receiver}
+	})
+}
+
+// listReceivers lists the IDs of all the processors found in pipelines
+func listProcessors(pipelines map[pipeline.ID]PipelineConfig) []component.ID {
+	return listComponents(pipelines, func(c PipelineConfig) []component.ID {
+		return c.Processors
+	})
+}
+
+type rawYAMLConfig struct {
+	Receivers  map[component.ID]yaml.Node     `mapstructure:"receivers"`
+	Processors map[component.ID]yaml.Node     `mapstructure:"processors"`
+	Pipelines  map[pipeline.ID]PipelineConfig `mapstructure:"pipelines"`
+}
+
+func (c *rawYAMLConfig) validate() error {
+	if len(c.Pipelines) == 0 {
+		return errors.New("missing pipelines")
+	}
+	for _, pipeline := range c.Pipelines {
+		if pipeline.Receiver != nil {
+			if _, found := c.Receivers[*pipeline.Receiver]; !found {
+				return fmt.Errorf("receiver %q not defined", pipeline.Receiver.String())
+			}
+		}
+
+		for _, processor := range pipeline.Processors {
+			if _, found := c.Processors[processor]; !found {
+				return fmt.Errorf("processor %q not defined", processor.String())
+			}
+
+		}
+	}
+
+	return nil
+}
+
+const configProviderScheme = "config"
+
+func newConfmapProviderFactory(main rawYAMLConfig) confmap.ProviderFactory {
+	return confmap.NewProviderFactory(createRawConfigProvider(main))
+}
+
+func createRawConfigProvider(main rawYAMLConfig) confmap.CreateProviderFunc {
+	return func(_ confmap.ProviderSettings) confmap.Provider {
+		return &rawConfigProvider{main: main}
+	}
+}
+
+type rawConfigProvider struct {
+	main rawYAMLConfig
+}
+
+func (p *rawConfigProvider) Retrieve(ctx context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
+	key := strings.TrimPrefix(uri, configProviderScheme+":")
+
+	switch key {
+	case "main":
+		return confmap.NewRetrievedFromYAML(p.buildMainYML())
+	case "receivers":
+		d, err := yaml.Marshal(p.main.Receivers)
+		if err != nil {
+			return nil, err
+		}
+		return confmap.NewRetrievedFromYAML(d)
+	case "processors":
+		d, err := yaml.Marshal(p.main.Processors)
+		if err != nil {
+			return nil, err
+		}
+		return confmap.NewRetrievedFromYAML(d)
+	case "pipelines":
+		d, err := yaml.Marshal(p.main.Pipelines)
+		if err != nil {
+			return nil, err
+		}
+		return confmap.NewRetrievedFromYAML(d)
+	default:
+		return nil, fmt.Errorf("%s not found", uri)
+	}
+
+}
+
+func (p *rawConfigProvider) buildMainYML() []byte {
+	var buf bytes.Buffer
+	if len(p.main.Receivers) > 0 {
+		fmt.Fprintf(&buf, "receivers: ${%s:receivers}\n", configProviderScheme)
```

Comment:
When looking at this output, I'm a bit confused.
```suggestion
		fmt.Fprintf(&buf, "receivers: ${config:receivers}\n")
```

The expected output is not defined, so I'm just confused. Would it make sense to print the number of receivers here instead of hard coding `config`?

Reply:
Actually `p.main` can be directly encoded, not sure what I was thinking about when writing this. Done in 093451c5cc23ead6a36fa595d26896444de0fd5d.